dwave/optimization/symbols.pyx

```diff
@@ -2225,6 +2269,77 @@ cdef class LP(Symbol):
             encoder.encode({arg.decode(): val for arg, val in self.ptr.get_arguments()})
         )
 
+    cdef Py_ssize_t _num_columns(self) except -1:
+        """The number of columns in the LP."""
+        if self.ptr.variables_shape().size() != 1:
```

Comment:
Just curious, why not use `assert` for these?

Reply:
`assert`s in Cython are treated as Python asserts. Which most people leave on all the time anyway. So I generally just do a full exception as a habit :shrug: I think either would be OK

Reply:
Yep, was just thinking about it as simply a stylistic thing, i.e. `assert` implies that it's for developers and the case should never happen